### PR TITLE
Dyn 5118 webview2 library

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Changes for replacing the WebBrowser component in Library by WebView2 component:
(**Please keep in mind that most of the files were copied from LibraryViewExtensionMSWebBrowser to LibraryViewExtensionWebView2, only around 9 files were changed, so if you want to see the real changes you will need to compare both folders with a tool like Meld or WinMerge**).
- I added a new .NET 4.8 VS Library Project that will display the Library using WebView2
- All the files from "LibraryViewExtensionMSWebBrowser\Handlers" were copied to "LibraryViewExtensionWebView2\Handlers" without modifications.
- All the files from "LibraryViewExtensionMSWebBrowser\ViewModels" were copied to "LibraryViewExtensionWebView2\ViewModels" without modifications.
- All the files from"LibraryViewExtension\web\library\resources" were copied to "LibraryViewExtensionWebView2\web\library\resources" without modifications.
- The next files were copied from "LibraryViewExtensionMSWebBrowser" without modifications: EventObserver.cs, LibraryViewCustomization.cs, LibraryViewToolTip.cs
- The files that were mainly modified/created are: LibraryViewController.cs, LibraryViewExtensionWebView2.csproj, ScriptingObject.cs, ViewExtension.cs, library.html, LibraryView.xaml

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Changes for replacing the WebBrowser component in Library by WebView2 component:

### Reviewers

@QilongTang

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
GIF showing the Library behavior in DynamoSandbox.
LibraryWebView2 Behavior

**Comment by RobertGlobant20** [comment]
GIF showing the behavior in DynamoRevit.
LibraryWebView2 Revit Behavior

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj [hunk 1]
```diff
@@ -285,11 +291,4 @@
       <LastGenOutput>Resources.en-US.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
-  <!-- Copy the library view extension manifest file to the default folder for Dynamo-->
-  <Target Name="AfterBuildOps" AfterTargets="Build">
-    <ItemGroup>
-      <ExtensionDefinition Include="Library_ViewExtensionDefinition.xml" />
-    </ItemGroup>
-    <Copy SourceFiles="@(ExtensionDefinition)" DestinationFolder="$(OutputPath)\viewExtensions\" />
-  </Target>
```
Comment (QilongTang) [inline src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:294]: I dont think you intend to touch this file correct? The view extension should not be under the output folder directly but in `viewExtensions` folder

File: src/LibraryViewExtensionWebView2/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+
+[assembly: AssemblyTitle("LibraryViewExtensionWebView2")]
+// The following GUID is for the ID of the typelib if this project is exposed to COM
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/Properties/AssemblyInfo.cs:7]: You may also want to add the same `[assembly: InternalsVisibleTo("ViewExtensionLibraryTests")]` in order to add tests later for this view extension

File: src/LibraryViewExtensionWebView2/Views/LibraryView.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿
+using System.Windows.Controls;
+
+using Dynamo.LibraryViewExtensionWebView2.ViewModels;
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/Views/LibraryView.xaml.cs:4]: Let's remove the empty lines

File: src/LibraryViewExtensionWebView2/Views/LibraryView.xaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+             xmlns:wv2="clr-namespace:Microsoft.Web.WebView2.Wpf;assembly=Microsoft.Web.WebView2.Wpf"
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:Dynamo.LibraryViewExtensionWebView2.Views"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+    <Grid x:Name="mainGrid">
+        <wv2:WebView2 Visibility="Hidden"/>
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/Views/LibraryView.xaml:11]: Just curious why the initial visibility is Hidden?

**Comment by QilongTang** [comment]
Good stuff, can you summarize the changes from these files
```
LibraryViewController.cs, LibraryViewExtensionWebView2.csproj, ScriptingObject.cs, ViewExtension.cs, library.html, LibraryView.xaml
```

File: src/LibraryViewExtensionWebView2/Views/LibraryView.xaml (hunk 4 above)
Comment (RobertGlobant20) [reply src/LibraryViewExtensionWebView2/Views/LibraryView.xaml:11]: @QilongTang because otherwise in the Dynamo Start page will show the Library (also in the LibraryViewExtensionMSWebBrowser was hidden by default).

Comment (QilongTang) [reply src/LibraryViewExtensionWebView2/Views/LibraryView.xaml:11]: Thanks, good to know

File: src/LibraryViewExtensionWebView2/Views/LibraryView.xaml.cs (hunk 3 above)
Comment (RobertGlobant20) [reply src/LibraryViewExtensionWebView2/Views/LibraryView.xaml.cs:3]: @QilongTang fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/13330/commits/dbbfffde0f9cbb0b64f134820ca012abfb1f12ad

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj [hunk 5]
```diff
@@ -9,6 +9,12 @@
     <RootNamespace>Dynamo.LibraryViewExtensionMSWebBrowser</RootNamespace>
     <AssemblyName>LibraryViewExtensionMSWebBrowser</AssemblyName>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <OutputPath>bin\Debug\</OutputPath>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <OutputPath>bin\Release\</OutputPath>
+  </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (RobertGlobant20) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:17]: @QilongTang I did similar changes like the ones you did for switch from cefSharp to WebBrowser component, I think is you don't specify an output path it will take the one from "$(SolutionDir)Config\CS_SDK.props" (it will be generating the dll files in Dynamo\bin\AnyCPU\Debug).
Please check the changes done for switching to WebBrowser:
https://github.com/DynamoDS/Dynamo/pull/11957/files

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj (hunk 1 above)
Comment (RobertGlobant20) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:294]: @QilongTang for this case I took as a base your PR for switching from cefSharp to WebBrowser, in this case I can see you inserted this lines in the LibraryViewExtensionMSWebBrowser.csproj file for copying the Library_ViewExtensionDefinition.xml to the viewExtensions folder, If I left this line both projects LibraryViewExtensionMSWebBrowser.csproj  and LibraryViewExtensionWebView2.csproj will be copying the same file two times, then depending which is compiled first is the one that will remain (so the wrong viewExtension for Library could be loaded), let me know if it makes sense to you.
This is the PR: https://github.com/DynamoDS/Dynamo/pull/11957/files

**Comment by RobertGlobant20** [comment]
This is a GIF showing that the interaction between the Library and the Packages guide was fixed.
LibraryPackagesGuide_Interaction

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj (hunk 1 above)
Comment (QilongTang) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:294]: OK gotcha, since we will obsolete and remove this view extension soon. This is good

Comment (RobertGlobant20) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:294]: @QilongTang yes, I will create a task for removing the old Library extensions.
- LibraryViewExtension
- LibraryViewExtensionMSWebBrowser.

**Comment by QilongTang** [comment]
Although this one looks good from review so far, I am holding on merging since release pipeline blocked for now

**Review by QilongTang — APPROVED** [review]
